Reduce sound buffer size in order to decrease sound latency.

simontoens opened the pull request:
This reduces the sound latency to a point where it is not really noticeable.  I have tested this on my iPhone 5 and my iPad Air 2 with a few games and also octamed.

I think the Amiga emulators I have used previously make this a user-configurable value ("sound buffer size").  I don't know if it makes sense to make this user configurable for iAmiga since the set of hardware iAmiga can run on is limited... probably not very important right now.

simontoens committed:
Reduce sound buffer size in order to decrease sound latency.

emufreak commented:
Hi Simon. Thats great. I'll test this and then release a new version..

CypherXG commented:
Hi Emufreak.
I would like to send you my Version of the iAmiga. Do you have an Email Adress?
I have some problems with iOS8. Since iOS8 the screen is incorrect and i can't correct it.
Otherwise my Amiga has come completely. FPS counter, many screens, Amiga Museum, etc.
Maybe you can help me.

Greets CypherXG

simontoens commented:
Hi CypherXG, why don't you put your version up on GitHub?  Makes it easy to share that way.

See this site for emufreak's contact info: http://www.iuae-emulator.net/

simontoens committed:
Don't call audio_reset twice when restoring a saved state.

m68k_go (m68k_cmm_intrf.cpp), which is called when restoring a previously saved state, ends up calling audio_reset twice.  The 2nd call causes the sound output to be broken when restoring a saved state.

The first audio_reset is called from restore_custom (custom.cpp), which is called from restore_state (savestate.cpp).
The callchain is: m68k_go->restore_state->restore_custom.

The second audio_reset is called from customreset (custom.cpp).
The callchain is: m68k_go->customreset->audio_reset.

Both of the functions that end up calling audio_reset are only called from m68k_go.  Removing the 2nd call to audio_reset fixes the broken sound when restoring a saved state, while not seemingly having any other bad effect, so I am removing that audio_reset call from customreset.

simontoens commented:
Hi @emufreak I added another fix to this pull request - this fixes the sound when restoring a saved state.  See the commit description for details.

emufreak commented:
Hi CypherXG. I'm looking forward to look at your version. This is exciting it looks like I'm not on my own anymore with this project :-). My Email is emufr3ak@icloud.com

However I also think the easiest and best solution would be for you to upload the code to github. Then we could easily merge our sources.

I know the Issue with IOS 8 and the emulation screen. Its fixed in the recent defender branch. I think most of the relevant changes are in iAmigaAppDelegate.mm. Again with Github you could easily push my changes to your source.

emufreak commented:
Just merged your push request. I'm looking forward to testing it

emufreak commented:
Just tested it. Works fine. I'll release a new Version. I'd like to mention you in the release note. Is this fine?

simontoens commented:
Sounds good thanks!
